src/sources/soundsourceproxy.cpp

```diff
@@ -593,9 +593,6 @@ void SoundSourceProxy::updateTrackFromSource(
         // once in the past. Only overwrite this information if
         // new data has actually been imported, otherwise abort
         // and preserve the existing data!
-        if (metadataImported.first != mixxx::MetadataSource::ImportResult::Succeeded) {
-            return; // abort
-        }
```

Comment:
I am unsure if it is a good idea to continue at this point. The local variable `trackMetadata` might contain invalid or partially modified data if the import didn't succeed.

Reinitializing it with `m_pTrack->readTrackMetadata(&trackMetadata)` would be a safe option in this case.